src/text_formatter/text_formatter.ts

```diff
@@ -57,42 +47,39 @@ export function textToListFormat(input: String) {
  * @return a string-array where each element is one word
  */
 export function sentenceIntoWordList(input: String) {
-	const separator = /[ ,]+/; // List all the values where the content of a sentence will get splitted: whitespace[ ] and comma[,]
-	const splittedSentenceIntoListOfWords = input.split(separator);
-	const temp = [];
+
+	const splittedSentenceIntoListOfWords = input.split(wordSeparator);
+	const listOfWords = [];
 	for (let index = 0; index < splittedSentenceIntoListOfWords.length; index++) {
-		// remove all elements(senteces) which are a empty String after splitting
-		if (splittedSentenceIntoListOfWords[index].trim() == "") {
+		if (splittedSentenceIntoListOfWords[index].trim() == "") {	// remove all elements(senteces) which are a empty String after splitting
 			splittedSentenceIntoListOfWords.splice(index, 1);
 		} else {
-			temp.push(splittedSentenceIntoListOfWords[index].trim());
+			listOfWords.push(splittedSentenceIntoListOfWords[index].trim());
 		}
 	}
```

Comment:
This whole loop could be: `const listOfWords = splittedSentenceIntoListOfWords.map(w => w.trim()).filter(w => w.length > 0);`. Damn easy to read. Relatively efficient as well.